ironfist/src/cpp/registry_prefs.h

```diff
@@ -0,0 +1,42 @@
+#ifndef REGISTRY_PREFS_H
+#define REGISTRY_PREFS_H
+
+#include <Windows.h>
+#include <string>
+
+template<typename T> T read_registry_pref(const std::string& pref_key);
+
+template<> DWORD read_registry_pref(const std::string& pref_key)
+	{
+	DWORD Type;
+	DWORD cbData = 4;
+	DWORD Data = (DWORD)-1;
+	LPCSTR SubKey = "SOFTWARE\\New World Computing\\Heroes of Might and Magic 2\\1.0";
+	HKEY hKey = nullptr;
+	if(RegCreateKeyA(HKEY_LOCAL_MACHINE, SubKey, &hKey) == ERROR_SUCCESS)
+		{
+		RegQueryValueExA(hKey, pref_key.c_str(), 0, &Type, (LPBYTE)&Data, &cbData);
+		}
+
+	return Data;
+	}
+
+template<typename T> bool write_registry_pref(const std::string& pref_key, const T& value);
+
+template<> bool write_registry_pref(const std::string& pref_key, const DWORD& value)

```

Comment:
C-strings are tricky, and I avoid them whenever possible.  In this case it doesn't matter, but in my opinion it would be beneficial to use them in other places in the code over C-strings, simply because there is a lot more room for error working with char*'s vs std::strings.